src/signals/incident-management/containers/IncidentDetail/components/LocationForm/index.js

```diff
@@ -46,57 +58,59 @@ class LocationForm extends React.Component { // eslint-disable-line react/prefer
+    this.form.controls.coordinates.setValue(newLocation.geometrie.coordinates.join(','));
   }
-  locationForm = FormBuilder.group({
+
+  form = FormBuilder.group({
     coordinates: ['', Validators.required],
-    location: this.props.incidentModel.incident.location
+    location: this.props.incident.location
   });
 
   handleSubmit = (event) => {
     event.preventDefault();
 
     this.props.onPatchIncident({
-      id: this.props.incidentModel.incident.id,
+      id: this.props.incident.id,
       type: 'location',
       patch: { location: { ...this.state.newLocation } }
     });
-
-    this.props.onClose();
   }
 
   render() {
-    const { incidentModel, onClose } = this.props;
+    const { patching, error, onClose } = this.props;
     return (
       <div className="location-form">
         <FieldGroup
-          control={this.locationForm}
+          control={this.form}
           render={({ invalid }) => (
             <form onSubmit={this.handleSubmit}>
               <div>
                 <FieldControlWrapper
                   render={HiddenInput}
                   name="coordinates"
                   display="Coordinates"
-                  control={this.locationForm.get('coordinates')}
+                  control={this.form.get('coordinates')}
                 />
 
                 <FieldControlWrapper
                   render={MapInput}
                   name="location"
-                  control={this.locationForm.get('location')}
+                  control={this.form.get('location')}
                   onQueryResult={this.onQueryResult}
                 />
 
-                {incidentModel.error ? <div className="notification notification-red" >
-                  {incidentModel.error && incidentModel.error.response && incidentModel.error.response.status === 403 ?
+                {error ? <div className="notification notification-red" >
+                  {error && error.response && error.response.status === 403 ?
                       'U bent niet geautoriseerd om dit te doen.' :
                       'De nieuwe locatie kon niet worden gewijzigd.'}
                 </div> : ''}
 
-                <button className="location-form__submit action primary" type="submit" disabled={invalid || incidentModel.patching.location}>Locatie opslaan</button>
-                <button className="location-form__cancel action secundary-grey" onClick={onClose}>Annuleren</button>
+                <button className="location-form__submit action primary" type="submit" disabled={invalid} data-testid="location-form-button-submit">
+                  <span className="value">Status opslaan</span>
+                  {patching.location ? <span className="working"><div className="progress-indicator progress-white"></div></span> : ''}
+                </button>
+                <button className="location-form__cancel action secundary-grey" onClick={onClose} data-testid="location-form-button-cancel">Annuleren</button>
```

Comment:
Please check the button elements that are used throughout this PR; all of them need an explicit type.

Reply:
gedaan en die we moeten later daar ASC Buttons van maken